### PR TITLE
RUBY-1411 Make fetch work with symbols

### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

File: lib/bson/document.rb [hunk 1]
```diff
@@ -34,6 +34,25 @@ module BSON
   # @since 2.0.0
   class Document < ::Hash
 
+    # Get a value from the document for the provided key. Can use string or
+    # symbol access, but the fastest will be to always provide a key that is of
```
Comment (saghm) [inline lib/bson/document.rb]: I'm having trouble finding the documentation for convert_key (which I assume is part of Hashie), but is this true even if we're always calling `convert_key`? I'd prefer to see some documentation stating that this is faster based on the type of the key in the hash before stating this in our docs.

Comment (p-mongo) [reply lib/bson/document.rb]: Symbols are always converted to strings, making strings faster. The data is also stored as strings I would imagine making the comment technically accurate, though definitely confusing to a user.

Comment (saghm) [reply lib/bson/document.rb]: Can we just change it to "using a string will be faster"?

Comment (p-mongo) [reply lib/bson/document.rb]: That is what I did.

**Pull request merged**